### DISCUSSION TITLE
Programming ESP32-C3-MINI via UART

### DISCUSSION THREAD

**Discussion by leotordo** [HomeSpan/HomeSpan#552] [Q&A]
Hello everybody.
Finally I've the prototype board con my desk. I mounted a ESP32-C3-MINI-1 ([https://www.digikey.it/short/720b3d2d](https://www.digikey.it/short/720b3d2d))

I write my code and tested in past weeks on a ESP32-WROOM board (and programmed via USB port).

Now I routed the ESP32-C3 UART (TXD0 and RXD0) to a USB/UART converter.

May I program my board in this configuration? (today i can't try for other jobs)
I know I've to put the module in Download boot with strapping pin.

There are some changes I've to do in my code (I started from Examples in HomeSpan library) to match ESP32-C3-MINI-1 functions or the code works on both (WROOM and C3) without modifications?

Thanks
Stefano

**Comment by leotordo** [comment]
Update.
I tried to program the ESP32-C3 via UART.
It works.
I've no error. But.....

On the Serial Monitor I can't reach the menu for Wifi setup....always the ESP32 sends this screenshot

![image](https://user-images.githubusercontent.com/121090098/230626910-669e0300-fa70-41fc-870f-54248d22a3be.png)


I do some error?

Is it correct ESP32C3 Dev Module?

How to reach Wifi Setup screenshot on Serial Monitor

Thank you very much
Stefano

**Reply by HomeSpan** [reply]
Did you type 'W' into the Serial Monitor input window?

**Reply by leotordo** [reply]
Yes but nothing happens.
I tried to change board to the first
![image](https://user-images.githubusercontent.com/121090098/230629568-d0af11ba-cdea-46c8-895d-1adac51140f7.png)
and on serial monitor all messages (like Welcome to Homekit) disapears.
I think is a "board problem"
Whats the board if I've no a board?
Now I've this configuration
USB/UART converterer.
ESP32 UART0 cabled to USB converter
two manual switches for Download Boot Mode (one is EN and the other is GPIO9). If I short EN, short GPIO8 then release EN I see that the EPS32 is wating download code
![image](https://user-images.githubusercontent.com/121090098/230630163-1f53d43a-3f6f-4894-bc8c-8a72a13b51a3.png)

But on re-start it doesn't go on setup menu

**Comment by leotordo** [comment]
may be the problem is that I've not setted "control pin"?

**Reply by HomeSpan** [reply]
What happens when you type 'W'.  Is the command ignored?  What happens if you type a different command, such as 's'.   Based on the above screenshot it looks like you are using the new Arudino IDE.  Please check that you have the Serial Monitor set for the correct baud AND that you selected either "New Line" or "Both NL & CR".   HomeSpan is looking for a "new line" character.

**Reply by leotordo** [reply]
I tried W and other commands. Nothing

As you can see below the baud rate i 115200 (link in source code), and "New Line" is set

Note (green line) that it seems ESP32 reboot after sent line LedPin=17.....

may be some problem in my board. I can send you the schematics fi you want

![image](https://user-images.githubusercontent.com/121090098/230632927-0e3b746f-4ef5-42f5-9693-fd4763f5160a.png)

**Reply by HomeSpan** [reply]
You need to make sure you are not using a serial pin for the LED or for anything else.  I recommend you disconnect all LEDs or other connects from your board and run Example 1.  If that works, the problem is that you are choosing pins in your sketch that are interfering with the serial input.

**Reply by leotordo** [reply]
ok thanks. I'll try example 1 and a do a check of all connections. I'll back after the weekend. Thank you for suggestions
Stefano

**Comment by leotordo** [comment]
Example 1 works. good. Next week I do a deep hardware check
Have a nice w/end

![image](https://user-images.githubusercontent.com/121090098/230636332-f0e354ba-327f-4470-85a2-62b70f038806.png)

**Comment by leotordo** [comment]
Hello. Solved.
The problem was in my firmware. There was a wrong port number in DEV_DimmableLED. There was a WROOM GPIO that C3 had not.
Now the WROOM code works on C3 module
Thanks for support
Stefano
